mesh/attribute_sets.hpp

```diff
@@ -161,7 +161,8 @@ class AttributeSets
 
        The returned marker array will be of size @a max_attr and it will contain
        only zeroes and ones. Ones indicate which attribute numbers are present
-       in the @a attrs array.
+       in the @a attrs array. In the special case when @a attrs has a single
+       entry equal to -1 the @a marker array will contain all ones.
```

Comment:
When this was originally introduced, there was some debate about this functionality. My opinion is that it would be better to avoid this special case, since it is surprising behavior unless the user reads the documentation (it is not self-documenting). The convention is also arbitrary (why -1 and not 0, for example? both are invalid attributes). One option would be to add an additional static helper function that returns a marker array of all 1s, e.g.

```cpp
Array<int> AttributeSets::FullMarker(int max_attr)
{
   Array<int> marker(max_attr);
   marker = 1;
   return marker;
}
```

This would make the attribute-processing logic clearer at the call-site, rather than relying on a "magic value".

Reply:
Thanks, and yep - the current state looks good to me!